x-pack/plugins/osquery/server/handlers/action/create_action_handler.ts

```diff
@@ -88,20 +92,25 @@ export const createActionHandler = async (
       ? !!some(packSO?.references, ['type', 'osquery-pack-asset'])
       : undefined,
     queries: packSO
-      ? map(convertSOQueriesToPack(packSO.attributes.queries), (packQuery, packQueryId) =>
-          pickBy(
+      ? map(convertSOQueriesToPack(packSO.attributes.queries), (packQuery, packQueryId) => {
+          const dynamicQueryPresent = packQuery.query && containsDynamicQuery(packQuery.query);
+
+          return pickBy(
             {
               action_id: uuidv4(),
               id: packQueryId,
-              query: packQuery.query,
+              query:
+                dynamicQueryPresent && packQuery.query && ecsData
```

Comment:
I believe we do not need aditional `&& packQuery.query check`, since it is being checked in `dynamicQueryPresent`